File: ros2_ws/src/my_cpp_pkg/src/battery_monitor.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/srv/set_led.hpp"

class BatteryMonitorNode : public rclcpp::Node
{
public:
    BatteryMonitorNode() : Node("battery_monitor_node")
    {
        batteryState_ = 0;
        client_ = this->create_client<my_robot_interfaces::srv::SetLed>("set_led");   
        shortTimer_ = this->create_wall_timer(std::chrono::seconds(4), 
            std::bind(&BatteryMonitorNode::callbackShortTimer, this));
        longTimer_ = this->create_wall_timer(std::chrono::seconds(6), 
            std::bind(&BatteryMonitorNode::callbackLongTimer, this));
    }
private:
    bool batteryState_;  // 0 = empty, 1 = full
    rclcpp::Client<my_robot_interfaces::srv::SetLed>::SharedPtr client_;
    rclcpp::TimerBase::SharedPtr shortTimer_;
    rclcpp::TimerBase::SharedPtr longTimer_;

    void callbackShortTimer()
    {   
        batteryState_ = 0;

        while(!client_->wait_for_service(std::chrono::seconds(1)))
        {
            RCLCPP_INFO(this->get_logger(), "Standing here...");
        }

        auto request = std::make_shared<my_robot_interfaces::srv::SetLed::Request>();
        request->led_number = 3;
        request->led_status = batteryState_;
        
        client_->async_send_request(request, std::bind(&BatteryMonitorNode::callbackClient, this,
            std::placeholders::_1));

        // Stop 4 seconds timer
        shortTimer_->cancel();
        // Start 6 seconds timer
        longTimer_->reset();
    }

    void callbackLongTimer()
    {
        batteryState_ = 1;
    }

    void callbackClient(rclcpp::Client<my_robot_interfaces::srv::SetLed>::SharedFuture future)
    {

    }
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<BatteryMonitorNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/battery_monitor.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from my_robot_interfaces.srv import SetLed


class BatteryMonitorNode(Node):

    def __init__(self):
        super().__init__("battery_monitor_node")
        self.battery_state = False  # True = empty, False = full
        self.client_ = self.create_client(SetLed, "set_led")
        self.short_timer_ = self.create_timer(4.0, self.callback_short_timer)
        self.long_timer_ = self.create_timer(6.0, self.callback_long_timer)
        self.get_logger().info("battery_monitor node has been started")

    def callback_short_timer(self):
        self.battery_state = True

        while not self.client_.wait_for_service(1.0):
            self.get_logger().info("Standing here...")

        request = SetLed.Request()
        request.led_status = self.battery_state
        request.led_number = 3

        future = self.client_.call_async(request)
        future.add_done_callback(self.callback_service)

        self.short_timer_.cancel()
        self.long_timer_.reset()

    def callback_long_timer(self):
        self.battery_state = False

        while not self.client_.wait_for_service(1.0):
            self.get_logger().info("Standing here...")

        request = SetLed.Request()
        request.led_status = self.battery_state
        request.led_number = 3

        future = self.client_.call_async(request)
        future.add_done_callback(self.callback_service)

        self.long_timer_.cancel()
        self.short_timer_.reset()

    def callback_service(self, future):
        response = future.result()
        self.get_logger().info("Response was: " + str(response.success))

def main(args=None):
    rclpy.init(args=args)
    node = BatteryMonitorNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()